firmware/library/L1_Drivers/onboardled.hpp

```diff
@@ -7,14 +7,99 @@
 #pragma once
 
 #include <cstdint>
+#include "L1_Drivers/gpio.hpp"
+#include "L2_Utilities/macros.hpp"
 
 class OnBoardLedInterface {
  public:
-           virtual void Initialize()                                 = 0;
-           virtual void LedOn(uint8_t led_number)                    = 0;
-           virtual void LedOff(uint8_t led_number)                   = 0;
-           virtual void LedSet(uint8_t led_number, bool on)          = 0;
-           virtual void LedToggle(uint8_t led_number)                = 0;
-           virtual void LedSetAll(uint8_t value)                     = 0;
-           virtual uint8_t LedGetValues(void)                        = 0;
+  enum class LightState : bool
+  {
+      kOn = true,
+      kOff = false
+  };
+           virtual void Initialize()                                      = 0;
+           virtual void On(uint8_t led_number)                            = 0;
+           virtual void Off(uint8_t led_number)                           = 0;
+           virtual void Set(uint8_t led_number, LightState state)         = 0;
+           virtual void Toggle(uint8_t led_number)                        = 0;
+           virtual void SetAll(uint8_t value)                             = 0;
+           virtual uint8_t GetStates(void)                                = 0;
+};
+
+class OnBoardLed: public OnBoardLedInterface
+{
+ public:
+  // Initialize takes the array of Gpios, sets each one to an output, and
+  // then turns all of the leds by setting the output high.
+  void Initialize(void) override
+    {
+      for (uint8_t i = 0; i < 4; i++)
+      {
+        led[i].SetAsOutput();
+        led[i].SetHigh();
+      }
+    }
+
+  void On(uint8_t led_number) override
+  {
+    SJ2_ASSERT_WARNING(led_number > 3,
+      "Input Led number can't be greater than 3, input = %d.\n", led_number);
+    led[led_number].SetLow();
+  }
+
+  void Off(uint8_t led_number) override
+  {
+    SJ2_ASSERT_WARNING(led_number > 3,
+      "Input Led number can't be greater than 3, input = %d.\n", led_number);
+    led[led_number].SetHigh();
+  }
+
+  void Set(uint8_t led_number,
+     OnBoardLedInterface::LightState state = LightState::kOn) override
+  {
+    SJ2_ASSERT_WARNING(led_number > 3,
+      "Input Led number can't be greater than 3, input = %d.\n", led_number);
+    (state == LightState::kOn) ? On(led_number) : Off(led_number);
+  }
+
+  void Toggle(uint8_t led_number) override
+  {
+    SJ2_ASSERT_WARNING(led_number > 3,
+      "Input Led number can't be greater than 3, input = %d.\n", led_number);
+    bool swap = led[led_number].ReadPin();
+    (swap) ? Set(led_number, LightState::kOn):Set(led_number, LightState::kOff);
```

Comment:
Ternery statements are meant to return something. Prefer to use if and else statements rather than ternery statments, in cases where you are merely executing code. ternery's used in this way are less readable then if-else.